packages/amplify-codegen-e2e-tests/src/__tests__/env-codegen.test.ts

```diff
@@ -0,0 +1,37 @@
+import { 
+  createNewProjectDir, 
+  deleteProjectDir,
+  deleteProject,
+  addApiWithSchema,
+  initJSProjectWithProfile,
+  amplifyPush,
+  generateModels,
+  updateApiSchema
+} from "amplify-codegen-e2e-core";
+import { addEnvironment, checkoutEnvironment } from "../environment/env";
+const schema = 'simple_model.graphql';
+const schemaWithError = 'modelgen/model_gen_schema_with_errors.graphql';
+const apiName = 'envtest';
+
+describe('env codegen tests', () => {
+  let projectRoot: string;
+  beforeEach(async () => {
+    projectRoot = await createNewProjectDir('envCodegenTest');
+  });
+
+  afterEach(async () => {
+    await deleteProject(projectRoot);
+    deleteProjectDir(projectRoot);
+  });
+
+  it('should generate models in different environments', async () => {
+    await initJSProjectWithProfile(projectRoot, { envName: 'enva' });
+    await addApiWithSchema(projectRoot, schema, { apiName });
+    await amplifyPush(projectRoot);
+    await addEnvironment(projectRoot, { envName: 'envb' });
+    updateApiSchema(projectRoot, apiName, schemaWithError)
```

Comment:
Just a comment to explain the flow that is happening in this test would be nice, because we are testing a specific case when the modelgen in one environment fails and switching the environment succeeds.
TODO for later - Test case where modelgen succeeds in both environments to see if the previously generated files are cleared/replaced.